kafka/client_async.py

```diff
@@ -446,15 +446,33 @@ def _poll(self, timeout, sleep=True):
                 continue
             conn = key.data
             processed.add(conn)
+            did_read = False
             while conn.in_flight_requests:
                 response = conn.recv() # Note: conn.recv runs callbacks / errbacks
+                did_read = True
 
                 # Incomplete responses are buffered internally
                 # while conn.in_flight_requests retains the request
                 if not response:
                     break
                 responses.append(response)
 
+            if not did_read:
```

Comment:
I don't think the client or the connection objects are thread-safe generally, but maybe I'm wrong!